sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RemoveAggsThroughLeftSemiAntiJoin.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.optimizer
+
+import org.apache.spark.sql.catalyst.plans.{LeftAnti, LeftSemi}
+import org.apache.spark.sql.catalyst.plans.logical.{Aggregate, Join, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.catalyst.trees.TreePattern.{AGGREGATE, LEFT_SEMI_OR_ANTI_JOIN}
+
+/**
+ * Remove the aggregation from left semi/anti join if the same aggregation has already been done
+ * on left side.
+ */
+object RemoveAggsThroughLeftSemiAntiJoin extends Rule[LogicalPlan] {
```

Comment:
`Remove ... Through ..` sounds a little strange. Do we have a similar rule like `Remove ... Through ..`? Otherwise, can we change the name?

Reply:
Do you have a recommended name?

Reply:
Since this is not `pushdown-like` optimizer, I'd drop `Through`. This is a kind of no-op removal, isn't it?

Reply:
What about `RemoveRedundantAggregatesInLeftSemiAntiJoin`?

Reply:
Either `RemoveRedundantAggregatesInLeftSemiAntiJoin`/`RemoveRedundantAggsInLeftSemiAntiJoin` or `RemoveAggregatesInLeftSemiAntiJoin`/`RemoveAggsInLeftSemiAntiJoin` are okay for me.

Reply:
Renamed it.